src/server/svr_jobfunc.c

```diff
@@ -5592,7 +5601,7 @@ svr_setjob_histinfo(job *pjob, histjob_type type)
 		 * exited with non-zero exit status, then it is FAILED,
 		 * otherwise FINISHED.
 		 */
-		newstate = JOB_STATE_FINISHED; /* default */
+		newstate = (pjob->ji_qs.ji_svrflags & JOB_SVFLG_SubJob) ? JOB_STATE_EXPIRED : JOB_STATE_FINISHED; /* default */
```

Comment:
Here also explain reason why this is done in comments.

Reply:
Since subjob never go to F state (instead go to X), so the newstate is initialized to X for subjob and F for others

Reply:
Can you update the same in comments. That helps people reading the code why it is done.

Reply:
Added in rc 13 https://github.com/PBSPro/pbspro/pull/590/commits/82b8f337b566ec556777394e4e3ef00fc07a0d4b